Frontend/src/app/clusters/[id]/page.tsx

```diff
@@ -0,0 +1,136 @@
+'use client';
+import React, { useEffect, useState } from "react";
+import { Descriptions, Input, Row, Col, Divider } from 'antd';
+import type { DescriptionsProps } from 'antd';
+import { Table, Tag } from 'antd';
+import type { TableProps } from 'antd';
+import { IClusterDetails } from "@/types/cluster-types";
+import { useAppSelector } from "@/redux/hook";
+import { useDispatch } from "react-redux";
+import { set_Selected_Cluster } from "@/redux/features/clusterData";
+import { getCluster } from "@/services/cluster-service";
+
+interface DataType {
+  key: string;
+  nodeID: string;
+  IPaddress: string;
+  Status: boolean;
+  Role: string;
+  UpTime: number;
+}
+
+const columns: TableProps<DataType>['columns'] = [
+  {
+    title: 'NodeID',
+    dataIndex: 'nodeID',
+    key: 'nodeID',
+  },
+  {
+    title: 'IP Address',
+    dataIndex: 'IPaddress',
+    key: 'ip',
+  },
+  {
+    title: 'Status',
+    dataIndex: 'Status',
+    key: 'status',
+    render: (_, { Status }) => (
+      <>
+        {Status ? (
+          <Tag color={'green'}>
+            {"Active"}
+          </Tag>) : (
+          <Tag color={"volcano"}>
+            {"Inactive"}
+          </Tag>
+        )}
+      </>
+    ),
+  },
+  {
+    title: 'Role',
+    dataIndex: 'Role',
+    key: 'role',
+  },
+  {
+    title: 'Uptime',
+    dataIndex: 'UpTime',
+    key: 'uptime',
+    render: (text) => <p>{text} days</p>
+  }
+];
+
+const { TextArea } = Input;
+
+export default function ClusterDetails({ params }: { params: { id: string } }) {
+  const dispatch = useDispatch();
+  const [loading, setLoading] = useState<boolean>(true);
+  const { selectedCluster } = useAppSelector(state => state.clusterData)
+  const [clusterDetails, setClusterDetails] = useState<IClusterDetails | null>(selectedCluster);
+
+  const items: DescriptionsProps['items'] = [
+    {
+      key: '1',
+      label: 'Cluster ID',
+      children: clusterDetails?._id || "",
+    },
+    {
+      key: '2',
+      label: 'JasmineGraph Version',
+      children: '1.21.101',
+    },
+    {
+      key: '3',
+      label: 'Platform',
+      children: '-',
+    },
+  ];
+
+  const fetchClusterDetails = async () => {
+    try{
+      const res = await getCluster(params.id);
+      if(res.data){
+        setClusterDetails(res.data)
+      }
+      dispatch(set_Selected_Cluster(res.data))
+    }catch(err){
+      console.log("Failed to fetch cluster")
```

Comment:
Please put more details i the error message. What do you mean by cluster here?

Reply:
We can connect to multiple JasmineGraph servers through the UI, those are referred to as the cluster.